src/LogicTasks/Semantics/Decide.hs

```diff
@@ -164,9 +164,22 @@ partialGrade DecideInst{..} sol = do
       english "The solution must contain at least one index."
     )
 
-  preventIfSolutionExceedsTableSize (length sol) (getTable formula)
+  preventWithHint (any (\x -> x < 1 || x > tableLen) sol)
+    (translate $ do
+      german "Lösung enthält nur gültige Indizes?"
+      english "Solution only contains valid indices?"
+    )
+    (translate $ do
+      german "Die Lösung enthält mindestens einen ungültigen Index."
+      english "The solution contains at least one invalid index."
+    )
+
+  preventIfSolutionExceedsTableSize (length sol) table
```

Comment:
Given that by the preceding test we know that each element of `sol` will be a value from `1` to `tableLen`, this test here now can only fail when there are duplicate elements in `sol`, right?

Then there might as well be done this as a more informative check: rejecting duplicate list elements there.

Or, one could even accept duplicate elements and then just ignore the duplicates in the subsequent grading, by applying `nubOrd` beforehand.

Reply:
Turns out the latter (or a more clumsy version of it) is already done anyway: https://github.com/fmidue/logic-tasks/blob/d1dbc54e2dfa3100f37de528d24ca08a95c2becb/src/LogicTasks/Semantics/Decide.hs#L225-L228

Reply:
I removed the test for duplicate elements.